src/script/interpreter.cpp

```diff
@@ -1964,7 +2298,9 @@ static bool ExecuteWitnessScript(const Span<const valtype>& stack_span, const CS
                 return set_error(serror, SCRIPT_ERR_BAD_OPCODE);
             }
             // New opcodes will be listed here. May use a different sigversion to modify existing opcodes.
-            if (IsOpSuccess(opcode)) {
+            if (is_vault_active && (opcode == OP_VAULT || opcode == OP_VAULT_RECOVER)) {
+                continue;
+            } else if (IsOpSuccess(opcode)) {
```

Comment:
This violates the "each `SCRIPT_VERIFY_foo` flag is a soft-fork" convention (see #10699 and #26439): if you have a correct `OP_VAULT` invocation, it will be invalid with script-verify flags set to `DISCOURAGE_UPGRADABLE_NOPS` but valid with them set to `DISCOURAGE_UPGRADABLE_NOPS | VAULT`.

I think this should instead be:

```c++
    if (opcode == OP_VAULT || opcode == OP_VAULT_RECOVER) {
        if (flags & SCRIPT_VERIFY_DISCOURAGE_VAULT) {
            return set_error();
        } else if ((flags & SCRIPT_VERIFY_FAULT) == 0) {
            // retain OP_SUCCESS behaviour if soft-fork hasn't activated
            return set_success(serror);
        }
    } else if (IsOpSuccess(opcode)) {
        ...
    }
```